Email verification

AMore080 opened the pull request:
email verification is finished. It makes use of the Resend API to send the email and will not allow the user to log in unless the token is verified which is going to filter out fake UH email addresses.

armaanamatya commented:
pls resolve the conflicts fast, its probably because of the new PRs I approved, pls rebase and try again, else we can talk more, good work tho

AMore080 commented:
cleaned up email verification email, rebased it so it should be good to merge now. Since it uses the resend api, we probably gotta check out vercel soon to start setting up all the ENV variables.
